File: binance4j-wallet/src/main/java/com/binance4j/wallet/dto/ApiTradingStatusTriggerCondition.java
package com.binance4j.wallet.dto;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

/**
 * Trading status trigger condition.
 * 
 * @param GCR  Number of GTC orders.
 * @param IFER Number of FOK/IOC orders.
 * @param UFR  Number of FOK/IOC orders.
 */
@ApiModel("Trading status trigger condition.")
public record ApiTradingStatusTriggerCondition(@ApiModelProperty("Number of GTC orders.") long GCR, @ApiModelProperty("Number of FOK/IOC orders.") long IFER,
		@ApiModelProperty("Number of FOK/IOC orders.") long UFR) {
}

File: connectors/src/main/kotlin/com/binance4j/wallet/dto/ApiTradingStatusTriggerCondition.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.wallet.dto

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

/**
 * Trading status trigger condition.
 *
 * @property gcr Number of GTC orders.
 * @property ifer Number of FOK/IOC orders.
 * @property ufr Number of FOK/IOC orders.
 */
@ApiModel("Trading status trigger condition.")
data class ApiTradingStatusTriggerCondition(
	@ApiModelProperty("Number of GTC orders.")
	val gcr: Long,
	@ApiModelProperty("Number of FOK/IOC orders.")
	val ifer: Long,
	@ApiModelProperty("Number of FOK/IOC orders.")
	val ufr: Long,
)
